src/ImageSharp/Common/Helpers/Vector128Utilities.cs

```diff
@@ -88,6 +99,60 @@ public static Vector128<float> ShuffleNative(Vector128<float> vector, [ConstantE
         return Vector128.Shuffle(vector, indices);
     }
 
+    /// <summary>
+    /// Shuffle 16-bit integers in the high 64 bits of <paramref name="value"/> using the control in <paramref name="control"/>.
+    /// Store the results in the high 64 bits of the destination, with the low 64 bits being copied from <paramref name="value"/>.
+    /// </summary>
+    /// <param name="value">The input vector containing packed 16-bit integers to shuffle.</param>
+    /// <param name="control">The shuffle control byte.</param>
+    /// <returns>
+    /// A vector containing the shuffled 16-bit integers in the high 64 bits, with the low 64 bits copied from <paramref name="value"/>.
+    /// </returns>
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public static Vector128<short> ShuffleHigh(Vector128<short> value, [ConstantExpected] byte control)
+    {
+        if (Sse2.IsSupported)
+        {
+            return Sse2.ShuffleHigh(value, control);
+        }
+
+        // Don't use InverseMMShuffle here as we want to avoid the cast.
+        Vector64<short> indices = Vector64.Create(
+            (short)(control & 0x3),
+            (short)((control >> 2) & 0x3),
+            (short)((control >> 4) & 0x3),
+            (short)((control >> 6) & 0x3));
+
+        return Vector128.Create(value.GetLower(), Vector64.Shuffle(value.GetUpper(), indices));
```

Comment:
A better fallback would be to do this:
```suggestion
         Vector128<short> indices = Vector128.Create(
            0,
            1,
            2,
            3,
            (short)((control & 0x3) + 4),
            (short)(((control >> 2) & 0x3) + 4),
            (short)(((control >> 4) & 0x3) + 4),
            (short)(((control >> 6) & 0x3) + 4));

        return Vector128.Create(value.GetLower(), Vector64.Shuffle(value.GetUpper(), indices));
```

The reason is that `V64` isn't accelerated on all platforms and you already have a `V128`, so this allows you to avoid decomposing the vector and doing more operations.